File: restore/client/src/game.cpp
#include "../include/game.h"

game::game():name(""),eventsPerUser(map<string,vector<Event>>()), subscriptionId(-1)
{}

game::game(string _name, map<string,vector<Event>> events, int subId): name(_name),eventsPerUser(events), subscriptionId(subId)
{
}

game::~game()
{
}

vector<Event> game::getEventsPerUser(string username){
    return eventsPerUser[username];
}


int game::getSubscriptionId(){
    return subscriptionId;
}

void game::subscribeUser(int id){
    subscriptionId = id;
}

void game::unsubscribeUser(){
    subscriptionId = -1;
    eventsPerUser.clear();
}

void game::addEvent(string userename, Event newEvent){
    if(eventsPerUser.find(userename) != eventsPerUser.end()){
        auto x = vector<Event>();
        eventsPerUser.emplace(userename,x);
    }
    eventsPerUser[userename].push_back(newEvent);

}

File: string to event bug/server/src/main/java/bgu/spl/net/impl/gamesUpdate/Game.java
package bgu.spl.net.impl.gamesUpdate;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Game {
   // todo : add games details
   String gameName;
   List<Event> events;
   //List<Integer> usersSubscribed;
   Map<Integer,Integer> usersConnectionsIdsAndSubscriptionIds;


   public Game(String name/*todo: add game details */) {
      gameName = name;
      events = new ArrayList<Event>();
      usersConnectionsIdsAndSubscriptionIds = new HashMap<Integer,Integer>();
   }

   public void subscribeNewUser(int connectionId, int subscriptionId){
      usersConnectionsIdsAndSubscriptionIds.put(connectionId,subscriptionId);
   }

   public void unsubscribeUser(int connectionId){
      usersConnectionsIdsAndSubscriptionIds.remove(connectionId);
   }

   public String getName(){
      return gameName;
   }

   public int getSubscriptionId(int connectionId){
      return usersConnectionsIdsAndSubscriptionIds.get(connectionId);
   }

   public Map<Integer,Integer> getSubscriptionUsers(){
      return usersConnectionsIdsAndSubscriptionIds;
   }

   public void addEvent(Event newEvent){
      events.add(newEvent);
   }
}
